gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreUtils.java

```diff
@@ -333,35 +334,39 @@ private static State getSerDeProps(SerDeInfo si) {
     return fieldSchemas;
   }
 
+  /**
+   * Returns a Deserializer from HiveRegistrationUnit if present and successfully initialized. Else returns null.
+   */
   private static Deserializer getDeserializer(HiveRegistrationUnit unit) {
-    Deserializer deserializer = null;
-
     Optional<String> serdeClass = unit.getSerDeType();
-    if (serdeClass.isPresent()) {
-      String serde = serdeClass.get();
-      HiveConf hiveConf = new HiveConf();
+    if (!serdeClass.isPresent()) {
+      return null;
+    }
 
-      try {
-        deserializer = ReflectionUtils.newInstance(hiveConf.getClassByName(serde).asSubclass(Deserializer.class),
-            hiveConf);
-      } catch (ClassNotFoundException e) {
-        LOG.warn("Serde class " + serde + " not found!", e);
-      }
+    String serde = serdeClass.get();
+    HiveConf hiveConf = new HiveConf();
 
-      if (deserializer == null) {
-        return deserializer;
-      }
+    Deserializer deserializer;
+    try {
+      deserializer = ReflectionUtils.newInstance(hiveConf.getClassByName(serde).asSubclass(Deserializer.class),
+          hiveConf);
+
+    } catch (ClassNotFoundException e) {
+      LOG.warn("Serde class " + serde + " not found!", e);
+      return null;
+    }
 
-      Properties props = new Properties();
-      props.putAll(unit.getProps().getProperties());
-      props.putAll(unit.getStorageProps().getProperties());
-      props.putAll(unit.getSerDeProps().getProperties());
+    Properties props = new Properties();
+    props.putAll(unit.getProps().getProperties());
+    props.putAll(unit.getStorageProps().getProperties());
+    props.putAll(unit.getSerDeProps().getProperties());
 
-      try {
-        SerDeUtils.initializeSerDe(deserializer, hiveConf, props, null);
-      } catch (SerDeException e) {
-        LOG.warn("Failed to initialize serde " + serde + " with properties " + props);
-      }
+    try {
+      SerDeUtils.initializeSerDe(deserializer, hiveConf, props, null);
+    } catch (SerDeException e) {
+      LOG.warn("Failed to initialize serde " + serde + " with properties " + props + " for table " + unit.getDbName() +
+          "." + unit.getTableName());
+      return null;
```

Comment:
This was the missing return statement. Without it, we might return an improperly initialized deserializer.